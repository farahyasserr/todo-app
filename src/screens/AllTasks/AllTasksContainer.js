import React, {useState, useCallback} from 'react';
import AllTasksView from './AllTasks';

const AllTasksContainer = () => {
  const [tasks, setTasks] = useState([
    {id: 1, title: 'Call mama', done: true},
    {id: 2, title: 'Call baba', done: false},
    {id: 3, title: 'Go shopping', done: true},
    {id: 4, title: 'Go To practice', done: false},
    {id: 5, title: 'Zaki', done: false},
  ]);

  const addNewTask = taskTitle => {
    setTasks(oldTasks => [
      ...oldTasks,
      {
        id: tasks.length + 1,
        title: taskTitle,
        done: false,
      },
    ]);
  };

  const setTaskAsDone = useCallback(
    taskId => {
      console.log('taaaaasks', tasks, taskId);
      const doneTaskIndex = tasks.findIndex(item => item.id === taskId);
      console.log('INDEX', doneTaskIndex);
      const newTasks = [...tasks];
      newTasks[doneTaskIndex].done = true;
      setTasks(newTasks);
    },
    [tasks],
  );

  const deleteTask = useCallback(
    taskId => {
      let taskIndex = tasks.findIndex(item => item.id === taskId);
      let newTasks = [...tasks];
      newTasks.splice(taskIndex, 1);
      setTasks(newTasks);
    },
    [tasks],
  );

  return (
    <AllTasksView
      tasks={tasks}
      addNewTask={addNewTask}
      setTaskAsDone={setTaskAsDone}
      deleteTask={deleteTask}
    />
  );
};

export default AllTasksContainer;
