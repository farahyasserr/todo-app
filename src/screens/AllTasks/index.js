import AllTasksScreen from './AllTasksContainer';

export default AllTasksScreen;
