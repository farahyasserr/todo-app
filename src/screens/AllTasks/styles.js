import {StyleSheet} from 'react-native';
import colors from '../../theme/colors';
export default StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: colors.grey,
  },
  listContainer: {
    paddingVertical: 16,
    paddingHorizontal: 16,
  },
  listContentContainer: {
    paddingVertical: 24,
  },
  buttonContainer: {
    paddingHorizontal: 16,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
