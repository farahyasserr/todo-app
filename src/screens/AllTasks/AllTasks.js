import React, {useState} from 'react';
import {SafeAreaView, View} from 'react-native';
import styles from './styles';
import {Button, Header} from '../../components';
import {AddTaskModal} from '../../blocks';
import TasksList from '../../blocks/TasksList';

const AllTasks = ({tasks, addNewTask, setTaskAsDone, deleteTask}) => {
  const [allTaskModalVisibility, setAllTaskModalVisibility] = useState(false);

  const setTaskAsDoness = id => setTaskAsDone(id);
  const onModalBackdropPress = () => setAllTaskModalVisibility(false);

  const onTaskSubmit = id => {
    addNewTask(id);
    setAllTaskModalVisibility(false);
  };

  const addTaskHandler = () => {
    setAllTaskModalVisibility(true);
  };

  return (
    <SafeAreaView style={styles.safeArea}>
      <Header title={'ALL_TASKS'} />
      <TasksList
        deleteTask={deleteTask}
        tasks={tasks}
        setTaskAsDone={setTaskAsDoness}
      />
      <View style={styles.buttonContainer}>
        <Button title={'ADD_TASK'} onPress={addTaskHandler} />
      </View>
      <AddTaskModal
        isVisible={allTaskModalVisibility}
        onBackDropPress={onModalBackdropPress}
        onTaskSubmit={onTaskSubmit}
      />
    </SafeAreaView>
  );
};

export default AllTasks;
