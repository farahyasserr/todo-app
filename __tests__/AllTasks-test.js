import React from 'react';
import renderer from 'react-test-renderer';
import AllTasks from '../src/screens/AllTasks/AllTasks';

jest.mock('react-native-modal', () => 'Modal');

test('renders correctly', () => {
  const tree = renderer.create(<AllTasks />).toJSON();
  expect(tree).toMatchSnapshot();
});
