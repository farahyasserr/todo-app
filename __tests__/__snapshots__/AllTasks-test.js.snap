// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 16,
        "opacity": 0,
        "paddingHorizontal": 16,
        "transform": Array [
          Object {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      ALL_TASKS
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/earth.png",
          }
        }
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "paddingVertical": 24,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 10,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <Text
        style={
          Object {
            "color": "#42aaf5",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        NO_TASKS
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "elevation": 5,
        "paddingHorizontal": 16,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#42aaf5",
          "borderRadius": 12,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 21,
            "lineHeight": 50,
          }
        }
      >
        ADD_TASK
      </Text>
    </View>
  </View>
  <Modal
    animationIn="zoomInDown"
    animationInTiming={600}
    animationOut="zoomOutUp"
    animationOutTiming={600}
    backdropOpacity={0.6}
    backdropTransitionInTiming={600}
    backdropTransitionOutTiming={600}
    isVisible={false}
    onBackdropPress={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 12,
          "padding": 22,
        }
      }
    >
      <TextInput
        autoFocus={true}
        onChangeText={[Function]}
        placeholder="ADD_TASK"
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "marginVertical": 8,
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          }
        }
        value=""
      />
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ddd",
            "borderRadius": 12,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "lineHeight": 50,
            }
          }
        >
          ADD
        </Text>
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
